src/bors/handlers/review.rs

```diff
@@ -0,0 +1,261 @@
+    notify_of_unapproval(&repo_state, pr).await
+}
+
+fn insufficient_approve_permission(repo: Arc<RepositoryState>, author: &GithubUser) -> bool {
+    !repo
+        .permissions
+        .load()
+        .has_permission(author.id, PermissionType::Review)
+}
+
+async fn deny_approve_request(
+    repo: &RepositoryState,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> anyhow::Result<()> {
+    tracing::warn!(
+        "Permission denied for approve command by {}",
+        author.username
+    );
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!(
+                "@{}: :key: Insufficient privileges: not in review users",
+                author.username
+            )),
+        )
+        .await
+}
+
+async fn notify_of_approval(
+    repo: &RepositoryState,
+    pr: &PullRequest,
+    approver: &str,
+) -> anyhow::Result<()> {
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!(
+                "Commit {} has been approved by `{}`",
+                pr.head.sha, approver
+            )),
+        )
+        .await
+}
+
+fn insufficient_unapprove_permission(
+    repo: Arc<RepositoryState>,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> bool {
+    author.id != pr.author.id
+        && !repo
+            .permissions
+            .load()
+            .has_permission(author.id, PermissionType::Review)
+}
+
+async fn deny_unapprove_request(
+    repo: &RepositoryState,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> anyhow::Result<()> {
+    tracing::warn!(
+        "Permission denied for unapprove command by {}",
+        author.username
+    );
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!(
+                "@{}: :key: Insufficient privileges: not in review users",
+                author.username
+            )),
+        )
+        .await
+}
+
+async fn notify_of_unapproval(repo: &RepositoryState, pr: &PullRequest) -> anyhow::Result<()> {
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!("Commit {} has been unapproved", pr.head.sha)),
+        )
+        .await
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::tests::mocks::{default_pr_number, BorsBuilder, Permissions, User, World};
+
+    #[sqlx::test]
+    async fn default_approve(pool: sqlx::PgPool) {
+        let world = World::default();
+        world.default_repo().lock().set_config(
+            r#"
+[labels]
+approve = ["+approved"]
+"#,
+        );
+        BorsBuilder::new(pool)
+            .world(world)
+            .run_test(|mut tester| async {
+                tester.post_comment("@bors r+").await?;
+                assert_eq!(
+                    tester.get_comment().await?,
+                    format!(
+                        "Commit pr-{}-sha has been approved by `{}`",
+                        default_pr_number(),
```

Comment:
I think it's find to hardcode these values here to make the test simpler. If we ever change them in the future, the test will just fail.

Reply:
Ah the reason I'm not hardcoding the values in was I didn't know where those values come from, and this was my attempt to explain the origin of those values for future me 😅

Reply:
That's not a bad reason! I hope we could make the default values so obvious that this wouldn't be needed though, and it would be immediately obvious from the test that `1` or `default-user` etc. is a special value :)